File: Arrays and Vectors/PrintSubarrays.cpp
#include<bits/stdc++.h>
using namespace std;

void printSubarrays(vector<int> & arr) {
    int n = arr.size();
    
    for(int i=0; i<n; i++) {
        for(int j=i; j<n; j++) {
            for(int k=i; k<=j; k++) {
                cout<<arr[k]<<",";
            }
            cout<<endl;
        }
    }
}

int main() {
    vector<int> arr = {1,2,3,4,5,6,7};
    printSubarrays(arr);
}

File: Arrays and Vectors/PrintSubarrays.js
const printSubarrays = (arr) => {
  let n = arr.length;

  for (let i = 0; i < n - 1; i++) {
    for (let j = i + 1; j < n; j++) {
      for (let k = i; k < j; k++) {
        process.stdout.write(`${arr[k]},`);
      }
      console.log();
    }
  }
};

const arr = [5, 8, 10, 12, 16, 21, 31, 55];
printSubarrays(arr);
